src/OpenColorIO/ops/fixedfunction/FixedFunctionOpCPU.cpp

```diff
@@ -1177,15 +1282,412 @@ void Renderer_LUV_TO_XYZ::apply(const void * inImg, void * outImg, long numPixel
     }
 }
 
+namespace 
+{
+namespace ST_2084
+{
+    static constexpr double m1 = 0.25 * 2610. / 4096.;
+    static constexpr double m2 = 128. * 2523. / 4096.;
+    static constexpr double c2 = 32. * 2413. / 4096.;
+    static constexpr double c3 = 32. * 2392. / 4096.;
+    static constexpr double c1 = c3 - c2 + 1.;
+
+#if OCIO_USE_SSE2
+    const __m128 abs_rgb_mask = _mm_castsi128_ps(_mm_setr_epi32(0x7fffffff, 0x7fffffff, 0x7fffffff, 0));
+    const __m128 vm1 = _mm_set1_ps(m1);
+    const __m128 vm2 = _mm_set1_ps(m2);
+    const __m128 vm1_inv = _mm_set1_ps(1.0f / float(m1));
+    const __m128 vm2_inv = _mm_set1_ps(1.0f / float(m2));
+    const __m128 vc1 = _mm_set1_ps(c1);
+    const __m128 vc2 = _mm_set1_ps(c2);
+    const __m128 vc3 = _mm_set1_ps(c3);
+#endif
+} // ST_2084
+} // anonymous
+
+template<typename T>
+Renderer_PQ_TO_LINEAR<T>::Renderer_PQ_TO_LINEAR(ConstFixedFunctionOpDataRcPtr & /*data*/)
+    : OpCPU() 
+{
+}
+
+template<typename T>
+void Renderer_PQ_TO_LINEAR<T>::apply(const void *inImg, void *outImg, long numPixels) const 
+{
+    using namespace ST_2084;
+    const float *in = (const float *)inImg;
+    float *out = (float *)outImg;
+
+    for (long idx = 0; idx < numPixels; ++idx) 
+    {
+        // RGB
+        for (int ch = 0; ch < 3; ++ch)
+        {
+            float v = *(in++);
+            const T vabs = std::abs(T(v));
+            const T x = std::pow(vabs, T(1.) / T(m2));
+            const T nits = std::pow(std::max(T(0), x - T(c1)) / (T(c2) - T(c3) * x), T(1.) / T(m1));
+            // Output scale is 1.0 = 10000 nits, we map it to make 1.0 = 100 nits.
+            *(out++) = std::copysign(float(T(100.0) * nits), v);
+        }
+
+        // Alpha
+        *(out++) = *(in++);
+    }
+}
+
+template <typename T>
+Renderer_LINEAR_TO_PQ<T>::Renderer_LINEAR_TO_PQ(ConstFixedFunctionOpDataRcPtr& /*data*/)
+    : OpCPU()
+{
+}
+
+template <typename T>
+void Renderer_LINEAR_TO_PQ<T>::apply(const void* inImg, void* outImg, long numPixels) const
+{
+    using namespace ST_2084;
+    const float* in = (const float*)inImg;
+    float* out = (float*)outImg;
+
+    for (long idx = 0; idx < numPixels; ++idx)
+    {
+        // RGB
+        for (int ch = 0; ch < 3; ++ch)
+        {
+            float v = *(in++);
+            // Input is in nits/100, convert to [0,1], where 1 is 10000 nits. 
+            const T L = std::abs(v * T(0.01));    
+            const T y = std::pow(L, T(m1));
+            const T ratpoly = (T(c1) + T(c2) * y) / (T(1.) + T(c3) * y);
+            const T N = std::pow(ratpoly, T(m2));
+            *(out++) = std::copysign(float(N), v);
```

Comment:
Please add this comment to capture Nick's comment in Issue #1968.
// Note: the PQ value for zero is 0.836^78.84 = 7.36e-07 so there is a very small jump in the mirroring at zero. However, this is 20x smaller than a single 16-bit code value, so it is not visually significant.